libs/state/docs/snippets/model-and-viewModel-with-rx-tate.md

```diff
@@ -0,0 +1,259 @@
+- `IViewList` represents a model that we actually need in our component template.
+
+```
+interface IList {
+  id: string;
+  name: string;
+  items: string[];
+}
+```
+```
+interface IViewList {
+  name: string;
+  items: string[];
+  itemsTotal: number;
+  isOpen: boolean;
+}
+```
+As you can see they are a bit different. Not all `IList` data needed in the View and at the same time `IViewList` contains some additional properties that are not in the `IList`.
+
+### ListService
+```
+@Injectable({providedIn: 'root'})
+export class ListService {
+
+  constructor(private api: ListApiService) { }
+
+  get(): Observable<IList> {
+    return this.api.get();
+  }
+}
+```
+### ListComponent
+```
+@Component({
+  selector: "app-list",
+  templateUrl: "./list.component.html",
+  styleUrls: ["./list.component.css"]
+})
+export class ListComponent implements OnDestroy {
+  private destroy$ = new Subject();
+  viewList: IViewList;
+
+  constructor(private listService: ListService) {
+    this.listService
+      .get(this.id)
+      .pipe(takeUntil(this.destroy$))
+      .subscribe(({items, name}) => {
+        this.viewList = {
+          name,
+          items,
+          itemsTotal: items.length,
+          isOpen: true,
+          buttonLabel: 'Close'
+        };
+      });
+  }
+
+  ngOnDestroy() {
+    this.destroy$.next();
+  }
+
+  toggle(isOpen: boolean) {
+    this.viewList.isOpen = isOpen;
+    this.viewList.buttonLabel = isOpen ? 'Close' : 'Open';
+  }
+}
+```
+Main part here:
+- Model that we are getting from our API is different from model that we actually need in the template (View). We need to make additional transformation inside `subscribe()`.
+
+### List template
+```
+<section>
+  <h1>{{viewList.name}} ({{viewList.itemsTotal}})</h1>
+  <button (click)="toggle(!viewList.isOpen)">{{viewList.buttonLabel}}</button>
+  <ul *ngIf="viewList.isOpen">
+    <li *ngFor="let item of viewList.items">{{item}}</li>
+  </ul>
+</section>
+```
+
+# Using RxState to handle Model & ViewModel
+
+Lets take a look at 2 suggestions of handling Model & ViewModel with RxState. 
+
+## Option 1. Holding Model & ViewModel inside of component.
+In this example we will get rid of ListService and communicate with our app API layer directly inside of the component.
+
+### List component
+```
+@Component({
+  selector: "app-list",
+  templateUrl: "./list.component.html",
+  styleUrls: ["./list.component.css"],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class ListComponent extends RxState<IList> {
+  toggle$ = new Subject<boolean>();
+
+ viewModel$: Observable<IViewList> = combineLatest([
+    this.select("name"),
+    this.select("items"),
+    this.toggle$.pipe(startWith(true))
+  ]).pipe(
+    map(([name, items, isOpen]) => ({
+      name,
+      items,
+      itemsTotal: items.length,
+      isOpen,
+      buttonLabel: isOpen ? "Close" : "Open"
+    })),
+    distinctUntilSomeChanged(['itemsTotal', 'isOpen])
```

Comment:
maybe you take a look at the selectSlice operator? look in this PR: #119 there is a snippet about selectSlice and viewModel

Reply:
Yes, definitely will do.